src/figures/nofTradeLeader.cpp

```diff
@@ -48,17 +53,30 @@ void nofTradeLeader::Serialize(SerializedGameData& sgd) const
 
 void nofTradeLeader::GoalReached()
 {
+    noBase* nob = gwg->GetNO(goalPos);
+    RTTR_Assert(dynamic_cast<nobBaseWarehouse*>(nob));
+    nobBaseWarehouse* targetWarehouse = static_cast<nobBaseWarehouse*>(nob);
```

Comment:
You can use `checkedCast` instead of the assert above and the static_cast. I added that because we have this pattern quite often. There is also `boost::polymorphic_downcast` which is very similar.

Reply:
good point, i switched to checkedCast